sdk/src/main/scala/com/horizen/account/state/StateTransition.scala

```diff
@@ -46,22 +46,23 @@ class StateTransition(view: AccountStateView, messageProcessors: Seq[MessageProc
     }
   }
 
-  private def preCheck(msg: Message): Unit = {
+  private def preCheck(msg: Message, fakeMsg: Boolean): Unit = {
```

Comment:
I would prefer to have `isFakeMessage` as a property on the `Message` which gets set to `true` in `TransactionArgs.toMessage()` and `false` in `EthereumTransaction.asMessage()`. What do you think?

Reply:
Seems fine to me, will do